tool/site/server/src/routes/auth/callback.ts

```diff
@@ -0,0 +1,51 @@
+import Hapi from "@hapi/hapi"
+import {URL} from "url"
+import createOAuthClient from "../../oauthUtils"
+import {setSessionValue} from "../../services/session"
+import {getPrBranchUrl, getRegisteredCallbackUrl, parseOAuthState, prRedirectEnabled, prRedirectRequired} from "../helpers"
+
+export default {
+  method: "GET",
+  path: "/callback",
+  options: {
+    auth: false
+  },
+  handler: async (request: Hapi.Request, h: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
+
+    // Local
+    if (process.env.ENVIRONMENT?.endsWith("sandbox")) {
+      request.cookieAuth.set({})
+      h.state("Last-Token-Fetched", Math.round(new Date().getTime() / 1000).toString(), {isHttpOnly: false})
```

Comment:
is this a new key? i like what this is leading up to.. :)

Reply:
it's very new but in current epsat, at the moment we're using it to give you a session timer in top left and auto logout. but yes, it is leading to something better. refresh tokens incoming in future pr